app/src/main/java/org/wikipedia/views/SurveyDialog.kt

```diff
@@ -18,15 +27,52 @@ import org.wikipedia.history.HistoryEntry
 import org.wikipedia.settings.Prefs
 import org.wikipedia.util.DimenUtil
 import org.wikipedia.util.FeedbackUtil
+import org.wikipedia.util.UriUtil
 
 object SurveyDialog {
+    // @TODO: update the strings after product confirm them
+    fun showContributionSurveyDialog(
+        activity: AppCompatActivity,
+        @StringRes titleId: Int,
+        @StringRes messageId: Int,
+        @DrawableRes icon: Int? = null,
+        delayMillis: Long = 0,
+        isCancellable: Boolean = false
+    ) {
+        activity.lifecycleScope.launch {
```

Comment:
May I know why we are using the `Coroutine` here?

Reply:
removed.